libraries/libfc/src/crypto/bls_private_key.cpp

```diff
@@ -25,38 +25,49 @@ namespace fc::crypto::blslib {
       return bls_private_key(v);
    }
 
-   template<typename Data>
-   Data from_wif( const string& wif_key )
+   string to_wif(vector<uint8_t> secret, const fc::yield_function_t& yield )
    {
-   /*   auto wif_bytes = from_base58(wif_key);
+      FC_ASSERT(secret.size() == 32);
+
+      std::array<uint8_t, 32> v2;
+
+      std::copy(secret.begin(), secret.end(), v2.begin());
+
+      const size_t size_of_data_to_hash = 32 + 1;
+      const size_t size_of_hash_bytes = 4;
+      char data[size_of_data_to_hash + size_of_hash_bytes];
+      data[0] = (char)0x80; // this is the Bitcoin MainNet code

```

Comment:
This seems ok to me, but curious what others think of this? My understanding is Bitcoin uses `0x80` for mainnet and `0xef` for testnets. This just corresponds to what we are doing for other `private_key` WIF encoding.

@arhag @spoonincode

Reply:
My first thought is whether these should ever be WIF encoded at all. We don't WIF encode R1 private keys -- they're always `PVT_R1...`. WIF encoding of the K1 private keys is legacy I believe (though `cleos create key` still does it :woman_shrugging:)

So would it make more sense to use a `PVT_BLS..` string only?

Reply:
I agree with @spoonincode. We do not need legacy encodings for the BLS key pairs. We should follow the encoding we use for R1 public/private keys: `PUB_BLS_...` and `PVT_BLS_...`. We don't even need an encoding for signatures as far as I know; but if we for some reason did, it would also follow the `SIG_BLS_...` encoding.